File: 2000 98 & ME/QuickStart/howto/samples/Services/MessageQueue/mqasync/vb/mqasync.vb
Imports System
Imports System.Messaging
Imports System.Threading

Class MQReceive
    Public Shared Sub Main()
        Dim appName As String = Environment.GetCommandLineArgs()(0)

        if(Environment.GetCommandLineArgs().Length <> 2) then
            Console.WriteLine("Usage: " + appName +" <queue>")
            Exit Sub
        End if

        Dim mqPath As String = ".\" + Environment.GetCommandLineArgs()(1)

        if(Not MessageQueue.Exists(mqPath))then
            Console.WriteLine("The queue '" + mqPath + "' does not exist!")
        Else
            Dim mq As MessageQueue = new MessageQueue(mqPath)
            Dim formatter As XmlMessageFormatter = CType(mq.Formatter,XmlMessageFormatter)
            formatter.TargetTypeNames = new String(){"System.String"}

            AddHandler mq.ReceiveCompleted, AddressOf OnReceiveCompleted
            mq.BeginReceive()
        End if

        Console.WriteLine("Press Enter to quit the sample")
        Console.ReadLine()
    end sub

    Public Shared Sub OnReceiveCompleted(source As Object, asyncResult As ReceiveAsyncEventArgs)
        Dim mq As MessageQueue = CType(source,MessageQueue)
        Dim m As Message = mq.EndReceive(asyncResult.AsyncResult)
        Console.WriteLine("Message: " + CStr(m.Body))
        mq.BeginReceive()
    end sub
end class

File: 2000 98 & ME/QuickStart/howto/samples/Services/MessageQueue/mqasync/mqasync.cs
using System;
using System.Messaging;
using System.Threading;

public class MQReceive {
    public static void Main(String[] args)
    {
        string appName = Environment.GetCommandLineArgs()[0];

        if ( args.Length != 1 ) {
            Console.WriteLine("Usage: " + appName +" <queue>");
            return;
        }

        string mqPath = ".\\" + args[0];

        if ( !MessageQueue.Exists(mqPath) ) {
            Console.WriteLine("The queue '" + mqPath + "' does not exist!");
        }
        else {
            MessageQueue mq = new MessageQueue(mqPath);
            ((XmlMessageFormatter)mq.Formatter).TargetTypeNames = new string[]{"System.String"};

            mq.ReceiveCompleted += new ReceiveCompletedEventHandler(OnReceiveCompleted);
            mq.BeginReceive();
        }

        Console.WriteLine("Press Enter to quit the sample");
        Console.ReadLine();
    }

    public static void OnReceiveCompleted(Object source, ReceiveAsyncEventArgs asyncResult){
        MessageQueue mq = (MessageQueue)source;
        Message m = mq.EndReceive(asyncResult.AsyncResult);
        Console.WriteLine("Message: " + (string)m.Body);
        mq.BeginReceive();
    }
}